src/message_dao_test.py

```diff
@@ -0,0 +1,115 @@
+        self.dao = message_dao.MessageDao('FAKE_GIT_PATH')
+        self.mock_connect.assert_called_once()
+        self.mock_commit.assert_called_once()
+        self.assertEqual(3, self.mock_execute.call_count)
+        self.reset_mocks(return_value=True, side_effect=True)
+        self.mock_connect.side_effect = RuntimeError("Shouldn't be called after init")
+
+    def reset_mocks(self, return_value=False, side_effect=False):
+        self.mock_connect.reset_mock(return_value=return_value, side_effect=side_effect)
+        self.mock_execute.reset_mock(return_value=return_value, side_effect=side_effect)
+        self.mock_commit.reset_mock(return_value=return_value, side_effect=side_effect)
+
+    def test_store(self):
+        # Testing storing Message that's not already in DAO 
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        sql_text = "REPLACE INTO Messages VALUES (%s, %s, %s, %s, %s, %s, %s)"
+        self.dao.store(email)
+        self.mock_execute.assert_called_once_with(sql_text, mock.ANY)
+        self.mock_commit.assert_called_once()
+        self.reset_mocks()
+
+        # Testing storing Message that's already in DAO 
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        self.dao.store(email)
+        self.mock_execute.assert_called_once()
+        self.mock_commit.assert_called_once()
+
+    @mock.patch.object(subprocess, 'check_output')
+    @mock.patch.object(message_dao, 'parse_message_from_str')
+    def test_get(self, mock_parse_msg, mock_check_output):
+        # Testing searching for a message id that's not in DAO
+        self.mock_commit.side_effect = RuntimeError("Shouldn't be called during get")
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        mock_parse_msg.return_value = email
+        self.mock_cursor.fetchone.return_value = (email.archive_hash, email.change_id)
+        self.assertEqual(email, self.dao.get('fake_message_id'))
+        self.assertEqual(2, self.mock_execute.call_count) 
+        self.reset_mocks()
+
+        # Testing searching for a message id that's not in DAO
+        self.mock_cursor.fetchone.return_value = None
+        self.assertIsNone(self.dao.get('not_in_dao'))
+        self.mock_execute.assert_called_once()
+
+    def test_size(self):
+        # Testing size of empty dao
+        self.mock_commit.side_effect = RuntimeError("Shouldn't be called during get")
+        self.mock_cursor.fetchone.return_value = (0,)
+        self.assertEqual(0, self.dao.size())
+        self.mock_execute.assert_called_once()
+        self.reset_mocks()
+
+        # Testing size of one dao
+        self.mock_cursor.fetchone.return_value = (1,)
+        self.assertEqual(1, self.dao.size())
+        self.mock_execute.assert_called_once()
+        self.reset_mocks()
+
+        # Testing size of many dao
+        self.mock_cursor.fetchone.return_value = (10,)
```

Comment:
as noted before, we only need one of these blocks.

Since we're using mocks, the code path covered by all of them is the exact same.
While the underlying sql lib might do something different if the table is empty, our code doesn't.

We just return what the mock gives us.

Let's truncate this test func down into just one block

Reply:
Yep, code coverage remains the same with just one block; removed the other two test cases.